### ISSUE TITLE
Draft MVP YouTube Partner Program

### ISSUE THREAD

**Issue by bedeho** [Joystream/marketing#258]
# Document

This is a draft, based on feedback and review that occurs in this issue, a new final issue will be created.

# Background

The original idea is described here

https://github.com/Joystream/marketing/issues/17

However, that initiative was foreseen to be deployed while on testnet, and also when the YT-synch infrastructure was deployed. Unfortunately, this infrastructure has been delayed, however, we can still achieve some of the goal of that initiative by launching a MVP YouTube Partner Program (YPP) which does not rely on this infrastructure. Later, we can incorporate this infrastructure when it becomes available.

# Proposal

## Overview

The key principles come from our experience running the incentives v3 program in our community, just radically simplified. The scheme can be summarised through the following breakdown of the life-cycle of a creator.

1. **Acquired:** A creator discovers the YPP through one among a variety of mechanisms:
   - Incentivised referral from a peer creator.
   - Direct outreach by Jsgenesis or marketing WG in DAO.
   - Social media sharing with embedded referral link.
2. **JS Channel Created:** Creator creates their channel through Atlas signup flow.
3. **YT Authenticated:** Creator initiates authentication flow where they prove that they own some channel on YouTube. When the creator authenticates, an entry is added to a Jsgenesis operated database, including a possible referrer channel, as well as their email. As part of this flow, the creator also has to sign off on the terms (see below). One important thing to highlight here is that they should be made aware that when YT-synch works, we will synch over their content automatically. When the channel is created, a special Jsgenesis membership is setup as a collaborator to facilitate such auto uploads later.
4. **Approved:** Either because you are authenticated, or because of manual intervention by Jsgenesis staff, the creator has been validated as genuine, and thus is able to participate in a number of rewarded activities. Once approved, a creator is approved forever, however, they may be suspended at a later time (see next). User receives an email about approval status updates. The purpose of the approval stage is in anticipation of a future where we do not allow users to directly start earning $JOY, because there could be risks of abuse, at this early stage, we should expect any user in the database to be automatically approved at the time of induction.
5. **Actions:** The creator engages in a set of actions which Jsgenesis has outlined as being rewarded in $JOY. The reward conditions will vary across actions, and also for the same action over time, as the program evolves in scale and scope. Examples actions include
   - Refer creator.
   - Publish video on Joystream.
   - Issue first NFT.
   - Share Atlas video on social media.
6. **Paid:** At relatively short periodic interval, Jsgenesis pays out creators who have engaged in different actions. These payouts largely are determined and executed manually, the database is updated to reflect who was paid for what, and a payout email is sent to each creator. Payouts are made out to the channel account.
7. **Suspended:** The creator, their conduct or conduct is found to be in violation of the program, as determined by Jsgenesis staff. This suspension may be lifted at a later time.

## YouTube Only?

The majority of this scheme, in terms of technicals, communication and terms, is directly about YT. This is because it has an automated way of authenticating channel ownership, and holds the largest single pool of video content creators. Having authenticated channel owners on Joystream means we can reduce later risk of abuse in the NFT and creator token markets, and we get to reduce risk of paying for videos where the publisher ends up not owning the proper full license for the media. **None the less, we should make room for creators from any platform, and we can do this by simply informing people to get in touch directly if they are not on YT, and doing manual verification in our database.**

## Example: Actions and Conditions

Payouts happen weekly, for the following actions with corresponding terms.

### 1. YT-verification

- Action: You prove to our infrastructure that you own a youtube channel, using Google authentication service.
- Terms:
    - Joystream channel avatar, cover, title and description is properly set.
    - YT channel is at least 3 months old.
    - YT channel has at least 10 videos, all published at least 1 month ago
    - YT channel has at least 50 followers
- Amount: Three tiers of $JOY amount based on range of followers.

### 2. Creator Referral

- Action: Some new creator gets verified with you as a referrer during authentication.
- Terms:
  - referrer has been approved.
  - new creator has been approved.
- Amount: 50% of whatever $JOY this creator unlocked through au

Such a referrer could be set automatically through cookies via some sort of sharing link the creator can generate.

### 3. Publishing Video

- Action: A video has been added to your Joystream channel.
- Terms:
    - channel has been approved
    - video is approved as genuine content by Jsgenesis staff, i.e. not white noise
    - video follows some content guideline pertaining to nudity, violence, etc.: **note, when Jsgenesis pays for content, the standard is set at a different level than mere publishing.
    - if video is from YT, there must be a link in the description.
- Amount: Three four, the first is just the baseline for any content, while the next three are sensitive to existing view counts on YT if content comes from this source.

### 4. YouTube Sharing

- Action: A Jsgenesis provided reference video is published under your YT channel. Analogous to this sort of announcement for a blockchain based twitter https://twitter.com/Matthew_Graham_/status/1542980576206012416.
- Terms:
    - video is publicly viewable with comments turned on.
    - video persists on channel for at least x days.
    - description has link to your Atlas channel.
- Amount: Three tiers again, based on how many followers channel has.

## Rules

We need some basic rules, similar to the rules in the [Founding Member Program](https://joystream.gitbook.io/testnet-workspace/testnet/founding-member-program#rules). Has to include stuff like them accepting that we may auto synch content when this starts working, they retain rights, but they may be evicted from program if abuse is identified.

## Future-proofing: Council Payouts

When the council payouts have been rolled out fully in `Ephesus`, we can use that mechanism as the means to actually do the final payouts in a way that integrates very nicely with Atlas. This means the creator gets the excellent benefits of having an integrated view of all their revenue, across NFTs, Jsgenesis incentives and whatever policy the DAO is choosing to adopt on top. We can get council buy-in by simply offering to have Jsgenesis finance the paying, by burning however much $JOY they are requesting the council will pay.

### A Preamble on Perverting Atlas!

A long standing, if not explicitly stated, principle behind Atlas product design has been that since it is a base upon which third parties are meant to build and iterate in the future, all Jsgenesis specific policies or initiatives should be kept outside of the product. I believe that it is time to break that principle in the interest of making the scheme described herein as effective as possible, and to instead focus on making it easy anyone configure, build and run Atlas with these customisations turned off. A silver lining here is also that it is not entirely inconceivable to see a useful role for other initial Atlas operators to run a similar scheme on behalf of the DAO.

### $JOY

In order to make calls to action effective, we should try to display explicit $JOY amount rewards, however this means we should also design for a future where there is a price feed where the $USD equivalent can be shown.

### Rebranding

The design changes to Atlas demanded by this scheme should be seen in light of other changes relating to the imminent separation of the Atlas product from the Joystream platform, described [here](https://github.com/Joystream/atlas/issues/2512).

#### Nudges

When the user is not logged in as a member with all channels approved, then show some sort of prominent "Join YPP" button in the top menu, this button triggers the YPP Page described next. So this means this is shown if user
- is not signed in at all
- is signed in with a user without any channels
- is signed in with a user with channels, but at least one is not in the program database.

#### YPP Page

This page is meant as not just first page of flow, but really main explainer of whole program, so it's critical. Later we can make a page on joystream.org perhaps, with leaderboards and metric etc, but not now. **The goal of the page is to get a creator to either try to connect to one of their existing YT accounts for verification, or if they are not on YT, to get them to get manually approved by getting in touch with staff.** The page must be very easy to link to and share, so the user should have the feeling that they can grab the URL from the address bar and get other people to the same page. The page also serves as the landing page for the referral linking described below, and it should display the channel avatar & title of whomever is currently the referrer if there is a cookie set for this. We have to assume that the creator knows very little when they hit this page, so we have to convey a number of concepts very quickly and clearly
- What we are asking them to do if they have YT channel:
   - create a channel on Joystream with same assets (title, avatar, cover photo, description...)
   - prove they own some YT channel using Google authentication
   - **allow us to replicate content over automatically in the near future.**
- What they are going to get for it:
   - a certain amount of $JOY
   - payouts happen every x days, but there can be delays.
   - ability to earn more $JOY by doing other actions, we should probably be rather specific here.
   - and we probably need to explain what $JOY is here

The call to action we should focus on is sending them into the YT-authentication flow, it is by far the most important, and the action to get in touch with staff should be a sidenote.

#### YT-authentication flow

As always, we probably should clearly be signalling how long this flow is, and what stage they are at.

1. User selects which JS channel they own, if they own more than one. If they don't have any, we should firs tell them to go make one for free. The user has to manually re-enter the flow if they had to leave to make a channel, but we can remind them of the opportunity at the end of the normal channel creation flow. The remainder of the flow is only possible once they have such a channel, and it is not already in the program.
2. Google authentication: look at this comparable service for another blockchain based music platform as a good reference for how that likely will look: https://youtubetoaudius.com/
3. User provides email where they will get notifications.
4. Show them the rules page where they accept various terms.
5. Show them success screen after backend has accepted.
6. Suggest they perform one among a variety of actions
   - refer a creator friend
   - publish a video
   - fix their channel presence

#### Normal channel creation flow

At the end of normal flow to create a channel, we should remind user of the opportunity to join YPP.

#### YT Studio

Adding some simple indicator about whether your channel is in the program, and whether it is approved or not. If it is, signal if it is suspended right now. If it you are approved we could also show

- link to current terms on some external notion page, which describes what is currently rewarded, and how.
- some sort of "refer a friend" link generation button.

## No Portal

We don't have an information portal thing, to see when is next payout, what you were paid, and why. Instead we just add a new Discord channel and say "ask here". If there is demand we can make a Retool dashboard based on the database later. The reason we do not invest in this portal now is because the payouts feature in Atlas will fully serve this purpose in the future.

### Emails

- Approval successful notice
- Payout notice

## Tech

We should have a simple AirTable database, and try to imitate key parts of the data model defined for incentives v3 here.

https://github.com/Joystream/marketing/issues/26

The model should be rich enough to capture concepts like
- channels, with referral and auth, approval, suspension status.
- rewarded actions

It makes sense to have reward payments being organised into rounds, as that makes it possible to aggregate data which is related in a clean way, for the purpose of having leaderboards and doign analysis. So introducing some sort of round numbering or something makes sense.

We also need APIs for things like
- listing channels
- reading state of a channel
- executing google authentication: use work here https://github.com/Joystream/youtube-synch

Updating the database in other ways should probably just be manual for now, then later one can construct dashboards or APIs if needed.

# Risks

Main risk here is being able to operate the program effectively with our current resources, it seems to require

- Support creators asking questions or having problems in some way.
- Monitor and maintain auth infrastructure and database
- Manually verify creators every few days.
- Sending out alert emails for people who got verified.
- Manually verify all actions of creators, which involves reviewing their Joystream presence, say identifying new videos or NFTs. Perhaps there is room for some scripts here.
- Computing payouts to everyone.
- Making payouts to everyone.
- Sending out alert emails for people who got paid.
- Adjust available actions and conditions.

We may need to hire someone part-time, perhaps from the community, to help with certain low risk steps here.

# Next Steps

- Feedback from key stakeholders.
  - Community
  - Atlas
  - Marketing: Robert
  - Operations: Martin
- Update scope and approach.
- Create explicit timeline for delivery.
- Create new issue for finalized plan.

**Comment by mrexpat** [comment]
Regarding creator referral rewards..

> Amount: 50% of whatever $JOY this creator unlocked through au 
>
So the referred person is unlocking some amount of $JOY based on how many followers they have, and the rewards we set, and the person who referred them earns on top of that?

**Example:**

_I invite MrBeast. He earns $10,000 worth of JOY due to his big following._
_Joystream rewards me with $5,000 worth of JOY, from a separate pool.. ($15,000 total)_

_Or MrBeast gets $5,000, and I as the referrer get the other $5,000 ($10,000 total)?_

I'm assuming it's the former..

--

Why are we doing this program?

- Is it to populate the platform with content?
- Is it a user acquisition strategy?

What is the hook of this program?

- Is it to get a copy of your youtube library on a decentralized video platform?
- Is it to earn $JOY by syncing your existing YT channel?

**Comment by bedeho** [comment]
> So the referred person is unlocking some amount of $JOY based on how many followers they have, and the rewards we set, and the person who referred them earns on top of that?

Yes.

> I'm assuming it's the former..

Yes.

> Why are we doing this program?

In order of importance

- build content catalogue
- (creator) user acquisition

>  What is the hook of this program?

if you want a single one-liner, I would say the second one.

**Comment by kdembler** [comment]
Interesting! Really like the idea, nothing technical strikes me as _tricky_ right away, seems like all should be doable. One concern that I have is the amount of manual work required from jsgenesis to make this program work. Before we rolled out the automatic tJOY withdrawal system, it seemed the amount of time spent by our staff on supporting the withdrawals was high and the amount of complaints from community on things not working was also high. With the scheme outlined here, our involvement would be substantial, with payouts being manual and all the creator actions having to be verified as genuine. Not saying we can't make this work, but we should be careful so this doesn't become the bottleneck of program's success

**Comment by dmtrjsg** [comment]
The Epic is now finished on the Backend side. The only thing that is not implemented is
